test/e2e/destroypod_test.go

```diff
@@ -212,3 +212,69 @@ func TestDestroyPodTimely(t *testing.T) {
 		t.Errorf("Time to delete pods = %v, want < %v", timeToDelete, revisionTimeout)
 	}
 }
+
+func TestDestroyPodWithRequests(t *testing.T) {
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "autoscale",
+	}
+	defer test.TearDown(clients, names)
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+
+	objects, err := v1a1test.CreateRunLatestServiceReady(t, clients, &names, v1a1opts.WithRevisionTimeoutSeconds(int64(revisionTimeout.Seconds())))
+	if err != nil {
+		t.Fatalf("Failed to create a service: %v", err)
+	}
+	domain := objects.Route.Status.URL.Host
+
+	pods, err := clients.KubeClient.Kube.CoreV1().Pods(test.ServingNamespace).List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", serving.RevisionLabelKey, objects.Revision.Name),
+	})
+	if err != nil || len(pods.Items) == 0 {
+		t.Fatalf("No pods or error: %v", err)
+	}
+
+	// The request will leep for more than 25 seconds.
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s?sleep=25001", domain), nil)
```

Comment:
Now that we have URL you should use the full URL instead of assuming the protocol.

Reply:
Actually I think we'd rather clean all the tests at once (since this pattern is throughout).

Reply:
Here is the issue where it is being done for conformance tests: https://github.com/knative/serving/issues/5030

Can you create another issue for e2e tests?

Reply:
Just for you: https://github.com/knative/serving/issues/5551